src/main/java/org/spongepowered/common/data/manipulator/mutable/entity/SpongeTameableData.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.mutable.entity;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.Ordering;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableTameableData;
+import org.spongepowered.api.data.manipulator.mutable.entity.TameableData;
+import org.spongepowered.api.data.value.mutable.OptionalValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.immutable.entity.ImmutableSpongeTameableData;
+import org.spongepowered.common.data.manipulator.mutable.common.AbstractData;
+import org.spongepowered.common.data.value.mutable.SpongeOptionalValue;
+import org.spongepowered.common.util.GetterFunction;
+import org.spongepowered.common.util.SetterFunction;
+
+import javax.annotation.Nullable;
+import java.util.UUID;
+
+public class SpongeTameableData extends AbstractData<TameableData, ImmutableTameableData> implements TameableData  {
+
+    @Nullable private UUID owner;
+
+    public SpongeTameableData(@Nullable UUID owner) {
+        super(TameableData.class);
+        this.owner = owner;
+        registerGettersAndSetters();
+    }
+
+    public SpongeTameableData() {
+        this(null);
+    }
+
+    @Override
+    public OptionalValue<UUID> owner() {
+        return new SpongeOptionalValue<UUID>(Keys.TAMED_OWNER, Optional.fromNullable(owner));
+    }
+
+    @Override
+    public TameableData copy() {
+        return new SpongeTameableData(this.owner);
+    }
+
+    @Override
+    public ImmutableTameableData asImmutable() {
+        return new ImmutableSpongeTameableData(this.owner);
+    }
+
+    @Override
+    public int compareTo(TameableData o) {
+        return ComparisonChain.start()
+                .compare(owner, o.owner().get().orNull(), Ordering.natural().nullsFirst())
+                .result();
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        return new MemoryDataContainer()
+                //TODO MemoryDataContainer will null check this.owner
+                .set(Keys.TAMED_OWNER.getQuery(), this.owner);
```

Comment:
I don't think you'd want to let the implementation throw an exception, if the owner is null, store an empty string. if it isn't, store the uuid. The builder should recognize this as well.